internal/daemon/daemon.go

```diff
@@ -147,7 +150,9 @@ func (d *Daemon) Run(ctx context.Context, listenPort string, stateDir string, so
 		return fmt.Errorf("Database recovery failed: %w", err)
 	}
 
+	d.extensionServersMu.Lock()
 	d.extensionServers = extensionServers
```

Comment:
Just to be safe, you should probably copy the supplied `extensionServers` into the daemon's struct because you modify the map down in `UpdateServers` which might have unintended consequences on the downstream project who supplied the map.

Reply:
Good catch thanks!

Reply:
Fixed